datafusion/physical-plan/src/joins/stream_join_utils.rs

```diff
@@ -740,20 +617,338 @@ pub fn record_visited_indices<T: ArrowPrimitiveType>(
     }
 }
 
+#[macro_export]
+macro_rules! handle_state {
+    ($match_case:expr) => {
+        match $match_case {
+            Ok(StreamJoinStateResult::Continue) => continue,
+            Ok(StreamJoinStateResult::Ready(res)) => Poll::Ready(Ok(res).transpose()),
+            Err(e) => Poll::Ready(Some(Err(e))),
+        }
+    };
+}
+
+#[macro_export]
+macro_rules! handle_async_state {
+    ($state_func:expr, $cx:expr) => {
+        $crate::handle_state!(ready!($state_func.poll_unpin($cx)))
+    };
+}
+
+/// Represents the possible results of a state in the join stream.
+pub enum StreamJoinStateResult<T> {
```

Comment:
Could you add some documentation about what the join would be "ready" for? I think it means "ready for output" but I am not sure 🤔

Reply:
It is used for simplifying the Poll results handling, nothing more. The API handles Poll::Ready and Poll::Pending conditions.